concrete/controllers/dialog/page/add/compose.php

```diff
@@ -93,7 +95,10 @@ public function submit()
                     $dateTime = new DateTime();
                     $publishDateTime = $dateTime->translate('cvPublishDate');
                     $publishEndDateTime = $dateTime->translate('cvPublishEndDate');
-                    if ($this->request->request->get('keepOtherScheduling')) {
+                    $app = Application::getFacadeApplication();
+                    $appConfig = $app->make(Repository::class);
+                    $liveVersionStatusOnScheduledVersionApproval = (string)$appConfig->get('concrete.misc.live_version_status_on_scheduled_version_approval');
+                    if (($liveVersionStatusOnScheduledVersionApproval === UNAPPROVED) xor (bool)$this->request->request->get('keepOtherScheduling')) {
```

Comment:
let's not use that constant. Also we don't use `xor` within our code so you'll have to rewrite this.